src/utils/validation/index.ts

```diff
@@ -27,35 +28,50 @@ export function isValidAssetID(assetText: string) {
 }
 
 // validation for send quantity inputs
-export function isValidQuantity(quantity: string) {
-  // ensure only a single decimal
-  if (quantity.split(".").length > 2) {
-    console.error("quantity entered has multiple decimals");
-    return false;
-  }
+export function isValidQuantity(quantity: string | BigNumber) {
+  quantity = bnToString(quantity);
 
   // ensure value is a string
   if (typeof quantity !== "string") {
     console.error("quantity entered is not a string");
     return false;
   }
 
+  // ensure only a single decimal
+  if (quantity.split(".").length > 2) {
+    console.error("quantity entered has multiple decimals");
+    return false;
+  }
+
   return true;
 }
 
+const bnToString = (value: BigNumber | string) => {
+  let converted = "";
+  if (BigNumber.isBigNumber(value)) {
+    converted = value.toString();
+    console.log(`converted ${value} to ${converted}`);
+    return converted;
+  }
+  console.log("returning original value, it was already a string:", value);
+  return value;
+};
+
 // further validation for sending functions, this is the last line of defense for
 // bad values finding their way into the "private" methods which require paying JUP
 // such as order placement, sending JUP, sending assets, etc...
 export function sendValidation(value: string): boolean {
-  console.log("validating quantity used for send:", value);
+  const quantity = bnToString(value);
+
+  console.log("validating quantity used for send:", quantity);
 
   // run the original validation function again as a sanity check
-  if (!isValidQuantity(value)) {
+  if (!isValidQuantity(quantity)) {
     return false;
   }
 
   // cannot be negative
-  if (parseInt(value) < 0) {
+  if (parseInt(quantity) < 0) {
```

Comment:
just be aware of these quirks, specifically item 1.

https://dmitripavlutin.com/parseint-mystery-javascript/

IMO, this code is probably fine seeing as any positive number is probably ok. we may eventually want to add decimal validations too, to ensure that decimals are rounded at the proper depth for the chain

Reply:
understood, IIRC there's like 100 different little gotchas like this with JS so I think we'll just have to address these as we go